### PR TITLE
Discussion Forum Prediction

### PR DESCRIPTION
**Pull request by aryantiwari10**
## Related Issue 

- Info about the related issue 

- [x] LGM-SOC'21 Participant 
- [ ] DevIncept Participant
- [x] Contributor

Closes: #555

### Describe the add-ons or changes you've made

Project Title : Discussion Forum Prediction

Dataset: https://www.kaggle.com/cdeotte/data-without-drift

Task : To predict about the discussion forums in detail.

In this project first we first performed a exploratory data analysis on the given dataset which includes of data cleaning , data manipulation, data preprocessing , data visualization and after that we used NLP for the stopwords removal and cleaning, after using NLP we did the model building using different machine learning classification and regression algorithms and then predicted the accuracy of every model . In the model prediction part we used different machine learning algorithms . In each algorithm we had included the accuracy score , training score , confusion matrix, classification report . While in the EDA part I have included different plots for the different visualizations of our dataset .I performed the ANN and feature scaling then, keeping epochs 200 and used different activation function and the optimizers during ANN prediction, During training and testing part we included classification models such as Decision Tree Classifer , Random Forest Classifier , Ada Boost Classifier model building . While the other model accuracies can be increased more using the hypertuning . Some plots which we used for visualizing the dataset are Histogram , Barplot , Boxplot, Heatmap , Scatter plot , Pairplot , Jointplot etc.

ACCURACIES OF DIFFERENT MODELS ARE:

KNeighbors Classifier= 60.02 %

Naiye Bayes= 49.12 %

Decision Tree Classifier= 56.8 %

Random Forest Classifier= 99.87 %

Logistic Regression : 62.27 %

Ada Boost Classifier= 53.25 %

LIBRARIES:

PANDAS

NUMPY

MATPLOTLIB

SEABORN

SCIPY

NLTK

SKLEARN

Keras

TensorFlow

CONCLUSION:

We got a good accuracy of about 99.87 % using Random Forest Classifier which is quite well for the given dataset.

The accuracy of other models can be increased further by HyperTuning.

-Aryan Tiwari

## Type of change

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Describe how it has been tested
Describe how have you verified the changes made

## Checklist:

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

## Screenshots (Only for Front End and UI/UX Designers)

 Original           | Updated
 :--------------------: |:--------------------:
 Original Screenshot | Updated Screenshot |

### CODE REVIEW

**Review by prathimacode-hub — APPROVED** [review]
Awesome, good to see your diversity in the choice of projects. Great going. Fab work. 🏆🙌@aryantiwari10

**Comment by aryantiwari10** [comment]
Thank you mam
